### PR TITLE
refactor(experimental): use new RPC libraries in @solana/accounts

### PR DESCRIPTION
**Pull request by lorisleiva**
This PR rewires the RPC dependencies of `@solana/accounts` to use the new RPC packages.

RPC packages reshuffle@2x (14)

### CODE REVIEW

File: packages/accounts/src/__tests__/__setup__.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
     AccountInfoWithBase58EncodedData,
     AccountInfoWithBase64EncodedData,
-    GetAccountInfoApi,
-    GetMultipleAccountsApi,
-    JsonParsedDataResponse,
-} from '../rpc-api';
+    SolanaRpcResponse,
+} from '@solana/rpc-types';
+
+import type { GetAccountInfoApi, GetMultipleAccountsApi, JsonParsedDataResponse } from '../rpc-api';
```
Comment (buffalojoec) [inline packages/accounts/src/__tests__/__setup__.ts:12]: Remind me again why we have to re-implement these in this package? It seems like through your refactoring, `@solana/rpc` doesn't need `@solana/accounts`, neither does `@solana/rpc-api`.

**Comment by lorisleiva** [comment]
✅ This PR was merged as part of https://github.com/solana-labs/solana-web3.js/pull/2168.

File: packages/accounts/src/__tests__/__setup__.ts (hunk 1 above)
Comment (buffalojoec) [reply packages/accounts/src/__tests__/__setup__.ts:12]: Yeahhhh I think we should hold off on that. That's like 60 packages! 🫠

Comment (lorisleiva) [reply packages/accounts/src/__tests__/__setup__.ts:12]: Agreed! 😅

**Pull request closed without merging**